### PR TITLE
Added feature extraction task documentation

### PR DESCRIPTION
**Pull request by akshay-babbar**
Hi, I have added documentation for feature-extraction task for [here](https://huggingface.co/tasks/feature-extraction). Please review since there are some things left still i guess in this but most of it is complete.

Thanks,
Akshay

### CODE REVIEW

File: tasks/src/feature-extraction/about.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## About the Task
+
+`Feature-Extraction` starts from initial set of measured data and builds derived values (features) intended to be informative and non-redundant, facilitating the subsequent learning and generalization steps in text, image & audio processing.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:3]:
```suggestion
Feature extraction task is to start from initial set of measured data, and build features intended to be informative and non-redundant, facilitating the subsequent learning and generalization steps in text, image & audio processing.
```
I would be in favor of even further simplifying the language 😅

File: tasks/src/feature-extraction/about.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+The Hub contains over 2000 models for [feature-extraction](https://huggingface.co/models?pipeline_tag=feature-extraction&p=2&sort=downloads).
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:4]:
```suggestion
```
I think we'd have to repeatedly update this so it's not needed, people can filter for themselves 🙂

File: tasks/src/feature-extraction/about.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Use Cases
+It is used for downstream  tasks like
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:7]:
```suggestion
Feature extraction can be used to transfer knowledge in natural language processing, computer vision and audio models.
```
I think this is somewhat simpler and less specific.

File: tasks/src/feature-extraction/about.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- Sentiment/Text classification
+- Image classification 
+- Audio Classification
+
+## Inference
+
+#### Feature Extraction
+
+```python
+## Loading the modules
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:18]:
```suggestion
```

File: tasks/src/feature-extraction/about.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+It is used for downstream  tasks like 
+- Unsupervised Learning
+- Sentiment/Text classification
+- Image classification 
+- Audio Classification
+
+## Inference
+
+#### Feature Extraction
+
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:16]: You can use `feature-extraction` pipeline like below 👇🏼

File: tasks/src/feature-extraction/about.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+```python
+## Loading the modules
+from transformers import pipeline
+#Fixing the chekpoint
+checkpoint = "facebook/bart-base"
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:21]:
```suggestion
```
also you haven't passed checkpoint in initialization?

File: tasks/src/feature-extraction/about.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+feature_extractor = pipeline("feature-extraction",framework="tf")
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:22]:
```suggestion
feature_extractor = pipeline(model = "facebook/bart-base", task = "feature-extraction")
```

File: tasks/src/feature-extraction/about.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+text = "\U0001F917 Transformers is an awesome library!"
+print(text)
+# 🤗 Transformers is an awesome library!
+feature_extractor(text,return_tensors = "tf")[0].numpy().mean(axis=0) ##Reducing along the first dimension to get a 768 dimensional array
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:26]:
```suggestion
# Reducing along the first dimension to get a 768 dimensional array
feature_extractor(text, return_tensors = "tf")[0].numpy().mean(axis=0) 
```
is it possible for you to write this in PyTorch, not sure if it would be simpler but it adds extra complexity for beginner + minimalistic tutorial IMHO

File: tasks/src/feature-extraction/about.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+#### Feature Extraction
+
+```python
+## Loading the modules
+from transformers import pipeline
+#Fixing the chekpoint
+checkpoint = "facebook/bart-base"
+feature_extractor = pipeline("feature-extraction",framework="tf")
+text = "\U0001F917 Transformers is an awesome library!"
+print(text)
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:24]:
```suggestion
```
I think this is not needed?

File: tasks/src/feature-extraction/about.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+#### Feature Extraction
+
+```python
+## Loading the modules
+from transformers import pipeline
+#Fixing the chekpoint
+checkpoint = "facebook/bart-base"
+feature_extractor = pipeline("feature-extraction",framework="tf")
+text = "\U0001F917 Transformers is an awesome library!"
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:23]:
```suggestion
text = "Transformers is an awesome library!"
```
Let's aim for simplicity!

File: tasks/src/feature-extraction/about.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+print(text)
+# 🤗 Transformers is an awesome library!
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:25]:
```suggestion
```

File: tasks/src/feature-extraction/about.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+feature_extractor(text,return_tensors = "tf")[0].numpy().mean(axis=0) ##Reducing along the first dimension to get a 768 dimensional array
+
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:27]: Can you print the output of call of feature_extractor?

File: tasks/src/feature-extraction/about.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+## Useful resources
+
+- [Feature-extractor](https://huggingface.co/docs/transformers/main_classes/feature_extractor)
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:32]:
```suggestion
- [Documentation for feature extractor of 🤗Transformers](https://huggingface.co/docs/transformers/main_classes/feature_extractor)
```

File: tasks/src/feature-extraction/data.ts [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+		},
+		{
+			description: "Pretrained weights for CodeBERT: A Pre-Trained Model for Programming and Natural Languages.",
+			id:          "microsoft/codebert-base",
+		},
+	],
+	spaces:       [
+
+	],
+	summary:      "Feature-Extraction starts from initial set of measured data and builds derived values (features) intended to be informative and non-redundant, facilitating the subsequent learning and generalization steps in text, image & audio processing.",
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/data.ts:47]:
```suggestion
	summary:      "",
```
It is same with the one in markdown and simpler language is appreciated 🙂 maybe take a look at the other task pages for inspiration in writing the summary? it's on top of each page!

File: tasks/src/feature-extraction/data.ts [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+	},
+	metrics: [
+		{
+			description: "",
+			id:          "",
+		},
+	],
+	models: [
+		{
+			description: "BART model pre-trained on English language. It was introduced in the paper BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension",
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/data.ts:36]:
```suggestion
			description: "A powerful feature extraction model for natural language processing tasks.",
```

File: tasks/src/feature-extraction/data.ts [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+			id:          "facebook/bart-base",
+		},
+		{
+			description: "Pretrained weights for CodeBERT: A Pre-Trained Model for Programming and Natural Languages.",
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/data.ts:40]:
```suggestion
			description: "A powerful feature extraction model for code language model.",
```

File: tasks/src/feature-extraction/data.ts [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+				content:
+						"India, officially the Republic of India, is a country in South Asia.",
+				type: "text",
+			},
+
+		],
+		outputs: [
+			{
+				table:
+						["array of shape (1,m,768) where m is the number of tokens in the input."][""],
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/data.ts:23]: Hmm, I don't know which way to represent this properly, pinging @osanseviero

**Review by merveenoyan — CHANGES_REQUESTED** [review]
I loved this! You put it well! 
I left couple of suggestions, can you check?

File: tasks/src/feature-extraction/data.ts (hunk 17 above)
Comment (osanseviero) [reply tasks/src/feature-extraction/data.ts:23]: Hmm maybe the table could have some dummy values, and the label could explain what it contains? WDYT?

File: tasks/src/feature-extraction/about.md (hunk 6 above)
Comment (akshay-babbar) [reply tasks/src/feature-extraction/about.md:21]: I have actually, so will pass this suggestion?

File: tasks/src/feature-extraction/about.md (hunk 12 above)
Comment (akshay-babbar) [reply tasks/src/feature-extraction/about.md:27]: It will be a 768 dimension tensor, would not that be too big?

File: tasks/src/feature-extraction/about.md (hunk 9 above)
Comment (akshay-babbar) [reply tasks/src/feature-extraction/about.md]: Sure

File: tasks/src/feature-extraction/about.md (hunk 7 above)
Comment (akshay-babbar) [reply tasks/src/feature-extraction/about.md]: I guess i am just using a custom model, so this should be fine?

**Comment by akshay-babbar** [comment]
Hey @merveenoyan ,thanks for the review , i have incorporated the changes with few comments, please review.

File: tasks/src/feature-extraction/data.ts (hunk 17 above)
Comment (merveenoyan) [reply tasks/src/feature-extraction/data.ts:23]: @akshay-babbar how does this sound?

Comment (akshay-babbar) [reply tasks/src/feature-extraction/data.ts]: Sorry i didn't see this and have added the entire 768 dim array to data.ts. Can you just have a look?
If the rendering of data.ts comes correct, i can replace it with a smaller dummy array or as you suggest.

**Comment by akshay-babbar** [comment]
@merveenoyan , For me `Numerical features` seems apt.

**Comment by merveenoyan** [comment]
@akshay-babbar does _embedding_ make sense? also want to ping @lewtun

**Comment by merveenoyan** [comment]
@akshay-babbar but numerical features is something can also be confused with an e.g. continuous column in dataframe (which is a very common convention) and is ambiguous IMO.

**Comment by akshay-babbar** [comment]
Okay @merveenoyan , also please let me know the remaining changes i should do to ensure the PR to get merged.

**Comment by merveenoyan** [comment]
@akshay-babbar it would be very appreciated if you could make the dataframe multi-dimensional (like a 2D one) as embeddings are often multidimensional and change the column name as embedding dimensions?

**Comment by akshay-babbar** [comment]
Hi @merveenoyan , In `types.ts` file, the tabular data has to be represented as `string [][]`, so i did precisely that but since the documents can not be build locally, i am not sure if the array is not coming as 2d though ideally it should.
Also i have changed the array's name to `embeddings`. Please check if it is getting reflected.

**Comment by akshay-babbar** [comment]
Hi @merveenoyan , is there an update for the above?

File: tasks/src/feature-extraction/data.ts [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+		inputs: [
+			{
+				label:   "Input",
+				content:
+						"India, officially the Republic of India, is a country in South Asia.",
+				type: "text",
+			},
+
+		],
+		outputs: [
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/data.ts:20]: Can you replace outputs with this instead? The table is too big so I can't do a commit suggestion.

outputs: [
			{
				label: "Embeddings",
				table:	[["Dimension 1", "Dimension 2", "Dimension 3"],
					["2.583383083343506", "2.757075071334839", "0.9023529887199402"],
					["8.29393482208252", "1.1071064472198486", "2.03399395942688"],
					["-0.7754912972450256", "-1.647324562072754", "-0.6113331913948059"],
					["0.07087723910808563", "1.5942802429199219", "1.4610432386398315"]],
				type: "tabular",
			},
		],

After that one it looks good to me!

Comment (akshay-babbar) [reply tasks/src/feature-extraction/data.ts:20]: Hi @merveenoyan I have included the above changes.

File: tasks/src/feature-extraction/about.md [hunk 19]
```diff
@@ -0,0 +1,35 @@
+## About the Task
+
+`Feature extraction` task is to start from initial set of measured data, and build features intended to be informative
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:3]:
```suggestion
Feature extraction is the task of building features intended to be informative from a given dataset,
```

File: tasks/src/feature-extraction/about.md [hunk 20]
```diff
@@ -0,0 +1,35 @@
+## About the Task
+
+`Feature extraction` task is to start from initial set of measured data, and build features intended to be informative 
+and non-redundant, facilitating the subsequent learning and generalization steps in text, image & audio processing.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:4]:
```suggestion
facilitating the subsequent learning and generalization steps in various domains of machine learning.
```

**Review by merveenoyan — APPROVED** [review]
LGTM!

File: tasks/src/Types.ts [hunk 21]
```diff
@@ -19,6 +19,7 @@ export type TaskDemoEntry = {
 	filename: string;
 	type: "img";
 } | {
+	label : string,
```
Comment (Pierrci) [inline tasks/src/Types.ts:22]: This is why the CI is failing in moon-landing @merveenoyan - this adds a mandatory `label` key for when `type: "tabular"`, but there's no such `label` in `table-question-answering`, `tabular-classification` and `tabular-regression`.

So either add a label in those cases or make it optional here (`label?: string;`) - you probably know better than me what's best :)
